src/proof/alethe/alethe_printer.cpp

```diff
@@ -0,0 +1,321 @@
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * The module for printing Alethe proof nodes.
+ */
+
+#include "proof/alethe/alethe_printer.h"
+
+#include <iostream>
+#include <unordered_map>
+
+#include "proof/alethe/alethe_proof_rule.h"
+
+namespace cvc5 {
+
+namespace proof {
+
+AletheProofPrinter::AletheProofPrinter()
+{
+  nested_level = 0;
+  step_id = 1;
+  prefix = "";
+  assumptions.push_back({});
+  steps.push_back({});
+}
+
+void AletheProofPrinter::alethePrinter(std::ostream& out,
+                                       std::shared_ptr<ProofNode> pfn)
+{
+  Trace("alethe-printer") << "- Print proof in Alethe format. " << std::endl;
+
+  // Special handling for the first scope
+  // Print assumptions and add them to the list but do not print anchor.
+  for (unsigned long int i = 3, size = pfn->getArguments().size(); i < size;
+       i++)
+  {
+    Trace("alethe-printer")
+        << "... print assumption " << pfn->getArguments()[i] << std::endl;
+    out << "(assume a" << std::to_string(i - 3) << " " << pfn->getArguments()[i]
+        << ")\n";
+    assumptions[0][pfn->getArguments()[i]] = i - 3;
+  }
+
+  // Then, print the rest of the proof node
+  alethePrinterInternal(out, pfn->getChildren()[0]);
+}
+
+std::string AletheProofPrinter::alethePrinterInternal(
+    std::ostream& out, std::shared_ptr<ProofNode> pfn)
+{
+  // Store current id in case a subproof overwrites step_id
+  int current_step_id = step_id;
+  // Special handling for the first scope
+  // Print assumptions and add them to the list but do not print anchor.
+  for (unsigned long int i = 3, size = pfn->getArguments().size(); i < size;
+       i++)
+  {
+    Trace("alethe-printer")
+        << "... print assumption " << pfn->getArguments()[i] << std::endl;
+    out << "(assume a" << std::to_string(i - 3) << " " << pfn->getArguments()[i]
+        << ")\n";
+    assumptions[0][pfn->getArguments()[i]] = i - 3;
+  }
+
+  // Then, print the rest of the proof node
+  alethePrinterInternal(out, pfn->getChildren()[0]);
+}
+
+std::string AletheProofPrinter::alethePrinterInternal(
+    std::ostream& out, std::shared_ptr<ProofNode> pfn)
+{
+  // Store current id in case a subproof overwrites step_id
+  int current_step_id = step_id;
+
+  // If the proof node is untranslated a problem might have occured during
+  // postprocessing
+  if (pfn->getArguments().size() < 3 || pfn->getRule() != PfRule::ALETHE_RULE)
+  {
+    Trace("alethe-printer")
+        << "... printing failed! Encountered untranslated Node. "
+        << pfn->getResult() << " " << toString(pfn->getRule()) << " "
+        << " / " << pfn->getArguments() << std::endl;
+    return "";
+  }
+
+  // Get the alethe proof rule
+  AletheRule vrule =
+      static_cast<AletheRule>(std::stoul(pfn->getArguments()[0].toString()));
```

Comment:
```suggestion
  AletheRule vrule = getAletheRule(args[0]);
```

Also use "arule", as the "v" is coming from the previous name.